manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -242,8 +243,16 @@ func (r *FybrikApplicationReconciler) checkReadiness(applicationContext Applicat
 			}
 			applicationContext.Application.Status.ProvisionedStorage[assetID] = provisioned
 			// register the asset
-			if newAssetID, err := r.RegisterAsset(assetID, dataCtx.Requirements.FlowParams.Catalog,
+			if response, err := r.RegisterAsset(assetID, dataCtx.Requirements.FlowParams.Catalog,
 				&provisioned, applicationContext.Application); err == nil {
+				newAssetID := response.AssetID
+				err = r.ValidateCreateAssetResponse(response, DataCatalogCreateAssetResponseTaxonomy, assetID)
+				if err != nil {
+					applicationContext.Log.Error().Err(err).Msg("failed to validate the catalog connector " +
+						"response when creating new asset")
+					setReadyCondition(applicationContext, assetID)
```

Comment:
We could not properly register the asset. Why is it ready?

Reply:
I'll fix that thanks